uiplib/gui/gallery.py

```diff
@@ -50,7 +52,13 @@ def show_image(self, image):
         self.tk_image = ImageTk.PhotoImage(show_image)
         self.cv.create_image(0, 0, anchor="nw", image=self.tk_image)
 
-    def blur_image(self, val):
-        """Blur chosen image by val amount."""
-        show = self.image.blur(val)
+    def _blur_helper(self, event):
+        """Helper to schedule the blur_image task."""
+        if self._job:
+            self.appObj.after_cancel(self._job)
+        self._job = self.appObj.after(250, self.blur_image)
```

Comment:
But, that's again making it too CPU intense @abhsag24 We need it to wait for some time before it does anything. It seems to be ideal at 200-250.